hyundai_kia_connect_api/VehicleManager.py

```diff
@@ -122,6 +122,8 @@ def stop_charge(self, vehicle_id: str) -> str:
         return self.api.stop_charge(self.token, self.get_vehicle(vehicle_id))
 
     def set_charge_limits(self, vehicle_id: str, ac: int, dc: int) -> str:
+        ac = int(float(ac), 0)
```

Comment:
What is the purpose of this line?
I have honestly never seen type casting with 2 arguments, I have no clue what it does.
But in any case, `ac` is already expected to be an int. Why cast it to float then back to int?
Same remark for `dc`

Reply:
I was finding in home assistant specifically sometimes a float was being passed.  What this does it ensures if 100.0 is passed it doesn't fail to set the charge limit which was happening prior.  float is a way to ensure we don't get an exception so:

```
value = 100.0
value = int(value)
```
Fails yet the above will succeed. 

As for the second value - that is a mistake.  I was trying to make it round off the decimals instead of the float approach.

Reply:
I see. In my opinion, it would be better to update home assistant to pass the correct format, if possible. A library should not compensate for the shortcoming of its dependencies.
If not possible, then I think we should at least update the type hint to "accept" other formats (Union)

> Fails yet the above will succeed.

That is strange. It works for me:

```
i>>> int(100.0)
100
>>> int(100.1)
100
```

Maybe home assistant is passing a string instead:
```
>>> int("100.0")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '100.0'
>>> int(float("100.0"))
100
```

Reply:
Agree on fixing on home assistant side, let's keep the library as lean as much as possible

Reply:
Home assistant was passing a float but the above code was just to cover all possibilities. I have updated this PR to remove this. I will update HA to ensure an int is always passed.